src/tty.rs

```diff
@@ -202,17 +203,66 @@ fn get_pw_entry(buf: &mut [i8; 1024]) -> Passwd {
     }
 }
 
+fn parse_shell(raw: Option<&String>) -> Option<Vec<CString>> {
```

Comment:
I think we will need something like this, at least for the configuration parsing. But I can move it to the config instead.